src/event_pusher/mod_event_pusher_rabbit.erl

```diff
@@ -88,102 +93,65 @@ push_event(Acc, _, _) ->
 %%% Internal functions
 %%%===================================================================
 
+-spec initialize_metrics(Host :: jid:server()) -> ok.
+initialize_metrics(Host) ->
+    [mongoose_metrics:ensure_metric(Host, Name, Type)
+     || {Name, Type} <- mongoose_rabbit_worker:list_metrics()].
+
 -spec create_exchanges(Host :: jid:server()) -> ok.
 create_exchanges(Host) ->
-    wpool:call(pool_name(Host), {create_exchanges, exchanges(Host)},
-               available_worker).
+    [wpool:call(pool_name(Host),
```

Comment:
We are silently ignoring fails here. If mongoose is not able to create exchange, there is no warning/crash.

Reply:
However it is okay if the worker crashes. But still the `mod_event_pusher_rabbit` is considered healthy. I think it would be ideally to prevent MongooseIM from starting, if we cannot declare exchanges.